packages/algob/src/lib/script-checkpoints.ts

```diff
@@ -182,8 +176,22 @@ export function persistCheckpoint (scriptName: string, checkpoint: Checkpoints):
   );
 }
 
+export function loadCheckpointByCPName (checkpointName: string): Checkpoints {
+  // Some structures are objects, some others are maps. Oh why.
+  const loaded = loadFromYamlFileSilent(checkpointName, { mapAsMap: true });
+  const obj: Checkpoints = {};
+  for (const [key, checkpointMap] of loaded.entries()) {
+    const cp: any = {};
+    for (const [cpKey, cpVal] of checkpointMap.entries()) {
+      cp[cpKey] = cpVal;
+    }
+    obj[key] = cp;
+  }
```

Comment:
so, why do we need to copy the `loaded` object?

Reply:
Because it's a map, not an object. YAML parser can either make everything an object or a map.
So `mapAsMap: true` sets this option to maps. And then I have to construct objects where needed.